test/README.md

```diff
@@ -0,0 +1,66 @@
+# Tests Overview
+
+## Full Invariant List
+
+1. `CoinbaseSmartWallet`
+   1. Only transaction path impacted is when `PermissionManager` is used as owner
+   2. Existing transaction paths work exactly the same
+      1. Can transact using other owners
+      2. Can remove `PermissionManager` as an owner
+      3. Can add and remove other owners
+      4. Can upgrade the contract
+2. `PermissionManager`
+   1. Can only validate signatures for ERC-4337 User Operations
+      1. Only supports the same UserOp type as `CoinbaseSmartWallet` (v0.6)
+   2. Cannot update `CoinbaseSmartWallet` owners
+   3. Cannot upgrade `CoinbaseSmartWallet` implementation
```

Comment:
these 2 invariants are protected by invariant 4 (cannot make self calls) right? If so, maybe explicitly saying the reason why `CoinbaseSmartWallet` cannot make self calls would be helpful. For example changing the wording of "as a precaution" to explicitly saying why it shouldn't make self calls